modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2259 @@
+      }
+    }
+    else if (_momentum_wall_types[bc_ind] == "symmetry")
+    {
+      {
+        std::string bc_type;
+        if (_porous_medium_treatment)
+          bc_type = "PINSFVSymmetryVelocityBC";
+        else
+          bc_type = "INSFVSymmetryVelocityBC";
+
+        InputParameters params = _factory.getValidParams(bc_type);
+        params.set<std::vector<BoundaryName>>("boundary") = {_wall_boundaries[bc_ind]};
+
+        MaterialPropertyName viscosity_name = _dynamic_viscosity_name;
+        if (_turbulence_handling != "none")
+          viscosity_name = NS::total_viscosity;
+        params.set<MaterialPropertyName>(NS::mu) = viscosity_name;
+
+        for (unsigned int d = 0; d < _dim; ++d)
+        {
+          NonlinearVariableName vname;
+          if (_porous_medium_treatment)
+          {
+            vname = NS::superficial_velocity_vector[d];
+            for (unsigned int d = 0; d < _dim; ++d)
+              params.set<MooseFunctorName>(u_names[d]) = NS::superficial_velocity_vector[d];
+            params.set<UserObjectName>("rhie_chow_user_object") = "pins_rhie_chow_interpolator";
+          }
+          else
+          {
+            vname = NS::velocity_vector[d];
+            for (unsigned int d = 0; d < _dim; ++d)
+              params.set<MooseFunctorName>(u_names[d]) = NS::velocity_vector[d];
+            params.set<UserObjectName>("rhie_chow_user_object") = "ins_rhie_chow_interpolator";
+          }
+
+          params.set<NonlinearVariableName>("variable") = vname;
+          params.set<MooseEnum>("momentum_component") = NS::directions[d];
+
+          _problem->addFVBC(bc_type, vname + "_" + _wall_boundaries[bc_ind], params);
+        }
+      }
+      {
+        const std::string bc_type = "INSFVSymmetryPressureBC";
+        InputParameters params = _factory.getValidParams(bc_type);
+        params.set<NonlinearVariableName>("variable") = NS::pressure;
+        params.set<std::vector<BoundaryName>>("boundary") = {_wall_boundaries[bc_ind]};
+
+        _problem->addFVBC(bc_type, NS::pressure + "_" + _wall_boundaries[bc_ind], params);
+      }
+    }
+  }
+}
+
+void
+NSFVAction::addINSEnergyWallBC()
+{
+  unsigned int skipped_energy_bc = 0;
+
+  for (unsigned int bc_ind = 0; bc_ind < _wall_boundaries.size(); ++bc_ind)
+  {
+    if (_energy_wall_types[bc_ind] == "fixed-temperature")
+    {
+      const std::string bc_type = "FVFunctionDirichletBC";
+      InputParameters params = _factory.getValidParams(bc_type);
+      params.set<NonlinearVariableName>("variable") = NS::T_fluid;
+      params.set<FunctionName>("function") = _energy_wall_function[bc_ind];
+      params.set<std::vector<BoundaryName>>("boundary") = {_wall_boundaries[bc_ind]};
+
+      _problem->addFVBC(bc_type, NS::T_fluid + "_" + _wall_boundaries[bc_ind], params);
+    }
+    else if (_energy_wall_types[bc_ind] == "heatflux" || _energy_wall_types[bc_ind] == "symmetry")
```

Comment:
```suggestion
    else if (_energy_wall_types[bc_ind] == "heatflux")
```

Reply:
symmetry is same as no BC so might as well not have a BC

Reply:
Yes, but what if a starting analyst wants to use symmetry and doesn't know that it is just a zero flux? I think I'll leave this option here. However, I'll add your changes below.

Reply:
Come to think about it, I think I'll go with your changes.

Reply:
Removed this guy, but not with the batch (same for the others below)